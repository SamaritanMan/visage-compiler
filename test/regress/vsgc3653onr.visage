/**
 * Regression test VSGC-3653 : Compiled bind: bound sequence: block
 *
 * On replace
 *
 * @test
 * @run
 */

class vsgc3653onr {
  var x = 5;
  var y = 50;
  def bs = bind {
              def low = bind x * 10;
              def high = bind y * 2;
              [low..<high]
           } on replace [a..b] = newVal { println("[{a}..{b}] = {sizeof newVal}") };
  function test() {
    x = 7;
    x = 0;
    y = 25;
    x = 2;
  }
}

vsgc3653onr{}.test()
