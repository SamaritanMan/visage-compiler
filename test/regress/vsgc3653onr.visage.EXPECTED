[0..-1] = 50
[0..19] = 0
[0..-1] = 70
[50..99] = 0
[0..19] = 0
